packages/ui-app/src/util/isValidBalance.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import BN from 'bn.js';
+import { IsValidWithMessage } from './types';
+import scientificNotationToNumber from './scientificNotationToNumber';
+
+// RegEx Pattern (positive int or scientific notation): http://regexlib.com/REDetails.aspx?regexp_id=330
+const reValidInputChars = RegExp('^[0-9\e\+\.]+[0-9\e\+\.]*$');
+
+export default function isValidBalance (input: any): IsValidWithMessage {
+  // always a string from <input type='number'> but leave as failsafe
+  if (!(typeof input === 'string')) {
+    throw Error('Balance input value must be of string type');
+  }
+
+  // impossible since using <input type='number'> and prevents spaces but leave as failsafe
+  input = input.toLowerCase().split(' ').join('');
+
+  const matchE = input.match(/e/gi);
+  const matchPlus = input.match(/\+/gi);
+  const matchEPlus = input.match(/e\+/gi);
+
+  const matchEBeforeNumberOrPlus = input.match(/(e)(?=[0-9+])/ig);
+  const matchEPlusBeforeNumber = input.match(/(e\+)(?=[0-9])/ig);
+
+  if (matchE && matchE.length > 1 || matchPlus && matchPlus.length > 1) {
+    return { isValid: false, errorMessage: 'Scientific notation may only contain one instance of \'e\' for scientific notation or exponential with \'e+\'' };
```

Comment:
- Each prop of the object on a seperate line.
- All messages need to be translatable, these are not

Reply:
(The above `t(errorMessage)` approach is not correct, it will just create havoc for a translation pass.)